### PR TITLE
Fix the final mypy --next errors

### PR DESCRIPTION
**Pull request by dabacon**
Will push changes to github actions after this is in.

* `np.einsum` is missing a type signature, resulting in some type ignores (filed issue at numpy https://github.com/numpy/numpy/issues/21978)
* In practice one can use numpy types like np.double for parameters, but currently we don't support that.  Filed #5758 to consider this.
* Numpy `tolist` does not return a type signature of `List`. wat?

Part of #3767

### CODE REVIEW

**Review by pavoljuhas — APPROVED** [review]
LGTM

File: cirq-core/cirq/linalg/transformations.py [hunk 1]
```diff
@@ -151,6 +151,7 @@ def targeted_left_multiply(
 
     all_indices = set(input_indices + data_indices + tuple(output_indices))
 
+    # TODO(#5757): remote type ignore when numpy has proper override signature.
```
Comment (mpharrigan) [inline cirq-core/cirq/linalg/transformations.py:154]: remove?

File: cirq-core/cirq/experiments/readout_confusion_matrix.py [hunk 2]
```diff
@@ -175,9 +175,11 @@ def _get_vars(self, qubit_pattern: Sequence['cirq.Qid']) -> List[int]:
     def _confusion_matrix(self, qubits: Sequence['cirq.Qid']) -> np.ndarray:
         ein_input = []
         for qs, cm in zip(self.measure_qubits, self.confusion_matrices):
-            ein_input += [cm.reshape((2, 2) * len(qs)), self._get_vars(qs)]
+            ein_input.extend([cm.reshape((2, 2) * len(qs)), self._get_vars(qs)])
         ein_out = self._get_vars(qubits)
-        ret = np.einsum(*ein_input, ein_out).reshape((2 ** len(qubits),) * 2)
+
+        # TODO(#5757): remote type ignore when numpy has proper override signature.
```
[the remark this answers is not in the document]
Comment (dabacon) [reply cirq-core/cirq/experiments/readout_confusion_matrix.py:181]: why?  this still has the problem with einsum?

[the remark this answers is not in the document]
Comment (mpharrigan) [reply cirq-core/cirq/experiments/readout_confusion_matrix.py:181]: you wrote "remote"

[the remark this answers is not in the document]
Comment (mpharrigan) [reply cirq-core/cirq/experiments/readout_confusion_matrix.py:181]: @dabacon

[the remark this answers is not in the document]
Comment (mpharrigan) [reply cirq-core/cirq/experiments/readout_confusion_matrix.py:181]: these typos appear to have been fixed in #5791

**Pull request merged**